src/Extensions/Logging/Internal/Replayable/OmexLogEventReplayer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Logging.Internal.Replayable;
+using Microsoft.Omex.Extensions.Logging.Replayable;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	internal class OmexLogEventReplayer : ILogEventReplayer
+	{
+		private readonly ILogEventSender m_sender;
+		private readonly IOptionsMonitor<OmexLoggingOptions> m_options;
+
+		public OmexLogEventReplayer(ILogEventSender sender, IOptionsMonitor<OmexLoggingOptions> options)
+		{
+			m_sender = sender;
+			m_options = options;
+		}
+
+		public bool IsReplayableMessage(LogLevel logLevel) =>
+			logLevel switch
+			{
+				LogLevel.Trace or LogLevel.Debug => m_options.CurrentValue.ReplayLogsInCaseOfError,
+				_ => false
+			};
+
+		public void AddReplayLog(Activity activity, LogMessageInformation logMessage) =>
+			activity.AddReplayLog(logMessage, m_options.CurrentValue.MaxReplayedEventsPerActivity);
+
+		public void ReplayLogs(Activity activity)
+		{
+			// Replay parent activity
+			if (activity.Parent != null)
+			{
+				ReplayLogs(activity.Parent);
+			}
+
+			foreach (LogMessageInformation log in activity.GetReplayableLogs())
+			{
+				string message = "[Replay] " + log.Message; // add prefix to mark log as replay
+				m_sender.LogMessage(activity, log.Category, LogLevel.Information, log.EventId, log.ThreadId, message, log.Exception);
```

Comment:
What would the log output be like here?
Will it do the same as the old replay logic - as in wrap the entire log metadata into the message field as a JSON representation - or is it extracting the metadata and converting all the logs to Info level?

Reply:
It would be different than what we had before. It will repeat the same log (tag, category, etc.) but adds the `[Replay]` prefix to the message to distinguish them.

Reply:
Oh, then there is a bug here.
If the level was higher than Info, it should get it's original level, rather than being downgraded to Info.

Feel free to fix it as a follow on item though.

Reply:
Replay stores only log messages lower then informational since others are already reported, so by filtering by activity id you would see all logs associated with it (replayed and regular)